fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnFE.java

```diff
@@ -515,6 +519,22 @@ public Expression visitCast(Cast cast, ExpressionRewriteContext context) {
         }
     }
 
+    // Check if the given literal value is safe to cast to the targetType.
+    // We need to guarantee FE cast result is identical with BE cast result.
+    // Otherwise, it's not safe.
+    protected boolean safeToCast(Cast cast) {
+        if (cast == null || cast.child() == null || cast.getDataType() == null) {
```

Comment:
if cast.child() is null, npe occurs in line 498("child.isNullLiteral()")
I think it would be better to throw an exception